parser/ParseCommand.cpp

```diff
@@ -0,0 +1,87 @@
+/**
+ *   Copyright 2016, Quickstep Research Group, Computer Sciences Department,
+ *     University of Wisconsin—Madison.
+ *
+ *   Licensed under the Apache License, Version 2.0 (the "License");
+ *   you may not use this file except in compliance with the License.
+ *   You may obtain a copy of the License at
+ *
+ *       http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *   Unless required by applicable law or agreed to in writing, software
+ *   distributed under the License is distributed on an "AS IS" BASIS,
+ *   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *   See the License for the specific language governing permissions and
+ *   limitations under the License.
+ **/
+
+#include "parser/ParseCommand.hpp"
+
+#include <cstdio>
+#include <map>
+#include <string>
+
+#include "parser/ParseString.hpp"
+#include "utility/StringUtil.hpp"
+
+#include "glog/logging.h"
+
+using std::string;
+using std::map;
+
+namespace quickstep {
+
+std::map<const std::string, std::string> ParseHelpCommand::help_messages_;
+
+void ParseHelpCommand::execute() {
+  // If the user did not enter any query text, display a generic message.
+  if (!query_) {
+    printf("Enter '.help {Sql Keyword} for usage information.'\n");
+    return;
+  }
+
+  // Look up the query in the map. If an exact match is not found, suggest
+  // the closest keyword.
+  const string lower_query = ToLower(query_->value());
+  const map<const string, string> &message_map = getHelpMessageMap();
+
+  // Try to find an exact match.
+  auto exact_itr = message_map.find(lower_query);
```

Comment:
Mark `exact_itr` as `const`.

Also, we typically use `it` to stand for an `iterator` instead of `itr`.

Finally, I'd suggest to use `matched_it` or similar to replace `exact_itr`.